Jinja2 rendering does not work when the loader is a ChoiceLoader

* pypugjs version:5.7.2
* Django version: N/A
* [Masonite](https://github.com/MasoniteFramework/masonite) version: 2.0.32 (using Jinja2)
* Python version: 3.6
* Operating System: Mac

### Description

I have a choice loader instead of a FileSystemLoader or PackageLoader but throws an exception here: https://github.com/kakulukia/pypugjs/blob/20c935811fdb26e0235561b4f04e449f0c593224/pypugjs/ext/jinja.py#L157

because choice loaders have no `searchpath` attribute (apparently).

```python
    def preprocess(self, source, name, filename=None):
        loader = self.environment.loader
        try:
            # we're in a Flask app
            loader = loader.app.jinja_loader
        except AttributeError:
            pass
>       if len(loader.searchpath):
E       AttributeError: 'ChoiceLoader' object has no attribute 'searchpath'
```

### What I Did

Here is a snippet of my code:

```python
            loader = ChoiceLoader(
                [PackageLoader(location[0], '/'.join(location[1:-1]))] + self.environments
            )

            self.env = Environment(
                loader=loader,
                autoescape=select_autoescape(['html', 'xml']),
                extensions=self._jinja_extensions
            )
```

this was throwing the exception above when rendering a `.pug` file:

What i did to fix it was simple set the searchpath to ''

```
            loader = ChoiceLoader(
                [PackageLoader(location[0], '/'.join(location[1:-1]))] + self.environments
            )

            # Set the searchpath since some packages look for this object
            # This is sort of a hack for now
            loader.searchpath = ''

            self.env = Environment(
                loader=loader,
                autoescape=select_autoescape(['html', 'xml']),
                extensions=self._jinja_extensions
            )
```

Paste the command(s) you ran and the output.
If there was a crash, please include the traceback here.

Here is a more full traceback:

```
self = <masonite.view.View object at 0x10eba7da0>

    def _render(self):
        try:
            # Try rendering the template with '.html' appended
>           return self.env.get_template(self.filename).render(
                self.dictionary)

masonite/view.py:87:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <jinja2.environment.Environment object at 0x10eb9add8>, name = 'pug/hello.pug.html', parent = None, globals = None

    @internalcode
    def get_template(self, name, parent=None, globals=None):
        """Load a template from the loader.  If a loader is configured this
            method asks the loader for the template and returns a :class:`Template`.
            If the `parent` parameter is not `None`, :meth:`join_path` is called
            to get the real template name before loading.

            The `globals` parameter can be used to provide template wide globals.
            These variables are available in the context at render time.

            If the template does not exist a :exc:`TemplateNotFound` exception is
            raised.

            .. versionchanged:: 2.4
               If `name` is a :class:`Template` object it is returned from the
               function unchanged.
            """
        if isinstance(name, Template):
            return name
        if parent is not None:
            name = self.join_path(name, parent)
>       return self._load_template(name, self.make_globals(globals))

venv/lib/python3.6/site-packages/jinja2/environment.py:830:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <jinja2.environment.Environment object at 0x10eb9add8>, name = 'pug/hello.pug.html'
globals = {'__pypugjs_attrs': <function attrs at 0x10ecd0ae8>, '__pypugjs_iter': <function iteration at 0x10ebe3840>, 'cycler': <class 'jinja2.utils.Cycler'>, 'dict': <class 'dict'>, ...}

    @internalcode
    def _load_template(self, name, globals):
        if self.loader is None:
            raise TypeError('no loader for this environment specified')
        cache_key = (weakref.ref(self.loader), name)
        if self.cache is not None:
            template = self.cache.get(cache_key)
            if template is not None and (not self.auto_reload or
                                         template.is_up_to_date):
                return template
>       template = self.loader.load(self, name, globals)

venv/lib/python3.6/site-packages/jinja2/environment.py:804:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <jinja2.loaders.ChoiceLoader object at 0x10eb9ad68>, environment = <jinja2.environment.Environment object at 0x10eb9add8>
name = 'pug/hello.pug.html'
globals = {'__pypugjs_attrs': <function attrs at 0x10ecd0ae8>, '__pypugjs_iter': <function iteration at 0x10ebe3840>, 'cycler': <class 'jinja2.utils.Cycler'>, 'dict': <class 'dict'>, ...}

    @internalcode
    def load(self, environment, name, globals=None):
        for loader in self.loaders:
            try:
                return loader.load(environment, name, globals)
            except TemplateNotFound:
                pass
>       raise TemplateNotFound(name)
E       jinja2.exceptions.TemplateNotFound: pug/hello.pug.html

venv/lib/python3.6/site-packages/jinja2/loaders.py:408: TemplateNotFound

During handling of the above exception, another exception occurred:

self = <test_view.TestView object at 0x10eba7cf8>

    def test_can_render_pubjs(self):
        view = self.container.make('ViewClass')
        view.add_extension('pypugjs.ext.jinja.PyPugJSExtension')
        assert view._jinja_extensions == ['jinja2.ext.loopcontrols', 'pypugjs.ext.jinja.PyPugJSExtension']

>       assert view.render(
            'pug/hello.pug', {'name': 'Joe'}).rendered_template == '<p>hello Joe</p>'

tests/test_view.py:236:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
masonite/view.py:80: in render
    self.rendered_template = self._render()
masonite/view.py:91: in _render
    return self.env.get_template(self.template).render(
venv/lib/python3.6/site-packages/jinja2/environment.py:830: in get_template
    return self._load_template(name, self.make_globals(globals))
venv/lib/python3.6/site-packages/jinja2/environment.py:804: in _load_template
    template = self.loader.load(self, name, globals)
venv/lib/python3.6/site-packages/jinja2/loaders.py:405: in load
    return loader.load(environment, name, globals)
venv/lib/python3.6/site-packages/jinja2/loaders.py:125: in load
    code = environment.compile(source, name, filename)
venv/lib/python3.6/site-packages/jinja2/environment.py:579: in compile
    source = self._parse(source, name, filename)
venv/lib/python3.6/site-packages/jinja2/environment.py:497: in _parse
    return Parser(self, source, name, encode_filename(filename)).parse()
venv/lib/python3.6/site-packages/jinja2/parser.py:40: in __init__
    self.stream = environment._tokenize(source, name, filename, state)
venv/lib/python3.6/site-packages/jinja2/environment.py:528: in _tokenize
    source = self.preprocess(source, name, filename)
venv/lib/python3.6/site-packages/jinja2/environment.py:522: in preprocess
    self.iter_extensions(), text_type(source))
venv/lib/python3.6/site-packages/jinja2/environment.py:521: in <lambda>
    return reduce(lambda s, e: e.preprocess(s, name, filename),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pypugjs.ext.jinja.PyPugJSExtension object at 0x10eb9a908>, source = 'p hello {{ name }}', name = 'pug/hello.pug'
filename = '/Users/joseph/Programming/core/resources/templates/pug/hello.pug'

    def preprocess(self, source, name, filename=None):
        loader = self.environment.loader
        try:
            # we're in a Flask app
            loader = loader.app.jinja_loader
        except AttributeError:
            pass
>       if len(loader.searchpath):
E       AttributeError: 'ChoiceLoader' object has no attribute 'searchpath'
```

--- a/pypugjs/ext/jinja.py
+++ b/pypugjs/ext/jinja.py
@@ -154,7 +154,7 @@ class PyPugJSExtension(Extension):
             loader = loader.app.jinja_loader
         except AttributeError:
             pass
-        if len(loader.searchpath):
+        if hasattr(loader, 'searchpath') and len(loader.searchpath):
             self.options["basedir"] = loader.searchpath[0]
 
         if not name or (name and not os.path.splitext(name)[1] in self.file_extensions):

